frontend/viewer/src/lib/components/ui/sidebar/sidebar-provider.svelte

```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+  import * as Tooltip from '$lib/components/ui/tooltip/index.js';
+  import {cn} from '$lib/utils.js';
+  import type {WithElementRef} from 'bits-ui';
+  import type {HTMLAttributes} from 'svelte/elements';
+  import {SIDEBAR_COOKIE_MAX_AGE, SIDEBAR_COOKIE_NAME, SIDEBAR_WIDTH, SIDEBAR_WIDTH_ICON} from './constants.js';
+  import {setSidebar} from './context.svelte.js';
+
+  let {
+    ref = $bindable(null),
+    open = $bindable(true),
+    onOpenChange = () => {},
+    class: className,
+    style,
+    children,
+    ...restProps
+  }: WithElementRef<HTMLAttributes<HTMLDivElement>> & {
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+  } = $props();
+
+  const sidebar = setSidebar({
+    open: () => open,
+    setOpen: (value: boolean) => {
+      open = value;
+      onOpenChange(value);
+
+      // This sets the cookie to keep the sidebar state.
+      document.cookie = `${SIDEBAR_COOKIE_NAME}=${open}; path=/; max-age=${SIDEBAR_COOKIE_MAX_AGE}`;
```

Comment:
this is really weird. The cookie is never being read so I'm not sure what it's for. It looks like the original ShadCN sidebar does the same thing. I say we either remove it or replace it with local storage.

Reply:
I think we should introduce a storage service asap to centralize this stuff. It can be backed by local-storage initially and then we improve it later.